pkg/storage/ingest/util.go

```diff
@@ -94,10 +96,6 @@ func commonKafkaClientOptions(cfg KafkaConfig, metrics *kprom.Metrics, logger lo
 		}),
 	}
 
-	if cfg.AutoCreateTopicEnabled {
-		opts = append(opts, kgo.AllowAutoTopicCreation())
-	}
```

Comment:
I think removing this cause another subtle change. I think `commonKafkaClientOptions()` is used by the generic `NewKafkaReaderClient()` and `NewKafkaWriterClient()`, so they're used for other topics too (e.g. block-builder). 

In this PR we both remove this code and we also changed the meaning of `auto-create-topic-default-partitions` (which doesn't change the Kafka default anymore, but it's just the number of partitions `createTopic()` will set), which means it will be a breaking change for other clients.

Probably the safest change would be preserving `setDefaultNumberOfPartitionsForAutocreatedTopics()` behaviour, preserving `kgo.AllowAutoTopicCreation()`, but force a topic creation too in our clients.

Alternative, I think a relatively safe change may be move `createTopic()` invokation to `NewKafkaReaderClient()` and `NewKafkaWriterClient()`.

Reply:
Discussed offline. Agree with Dimitar's change.